### DISCUSSION TITLE
Put quotes on Path name when installing a windows service with wix.

### DISCUSSION THREAD

**Discussion by altru123** [wixtoolset/issues#8168] [Questions]
Greetings,

Recently I noticed that the services which wix is installing are missing quotes in the PathName which is a security issue. I read that quotes should be added automatically when the path has space in it, but for some reason it creates the path name without quotes. I tried using the RegisterKey to rewrite the Path name, but it doesn't work, because the value is overwritten while wix is installing the windows service. Is there a simple way to add quotes to the path name? I am using wix v3, would an update fix the issue and start adding quotes to the services which I am installing?

Thank you in advance.

**Comment by altru123** [comment]
Additional note - the command that installs the windows service through wix is <ServiceInstall>. I think it is creating the Path Name registry value and overwriting my registry value with quotes. Is there a way to make my RegistryKey command execute after the ServiceInstall has finished executing.

**Comment by robmen** [comment]
> the services which wix is installing

The WiX Toolset does not install services. The Windows Installer installs services. The WiX Toolset builds the installation packages.

> I am using wix v3, would an update fix the issue

No. You need to update the Windows Installer. Microsoft is in charge of the Windows Installer.

**Reply by altru123** [reply]
Hello @robmen ,

Thank you for the reply. I was also wondering if it is possible to use Wix to add quotes to the "PathName" in the registry? Currently the windows installer creates the PathName registry entry without quotes and my <RegisterKey> command in my Wix file is being ignored or overwritten.

**Reply by robmen** [reply]
I don't know. I've never tried.
